src/Result.php

```diff
@@ -112,4 +143,112 @@ public function getMode()
     {
         return $this->parser->mode;
     }
+
+    /**
+     * Cleanup the relations on a models array
+     *
+     * @param $models
+     * @return array
+     */
+    public function cleanupRelationsOnModels($models)
+    {
+        $response = [];
+
+        if ($models instanceof Collection) {
+            foreach ($models as $model) {
+                $response[] = $this->cleanupRelations($model);
+            }
+        }
+
+        return $response;
+    }
+
+    /**
+     * Cleanup the relations on a single model
+     *
+     * @param $model
+     * @return mixed
+     */
+    public function cleanupRelations($model)
+    {
+        if (!($model instanceof Model)) {
+            return $model;
+        }
+
+        // get the relations which already exists on the model (e.g. with $builder->with())
+        $allowedRelations = $this->getRelationsRecursively($model);
+
+        // parse the model to an array and get the relations which got added unintentionally
+        // (e.g. when accessing a relation in an accessor method or somewhere else)
+        $response = $model->toArray();
```

Comment:
One issue I have with this is, that models are converted to arrays already when you want to get the result of your query. 
Usually you would get a Collection of Models, with this change you'll get an array of arrays. Can you think of a solution which doesn't require that change?